Assets/Plugins/iOS/Shopify/PaymentSession.swift

```diff
@@ -0,0 +1,221 @@
+    
+    let request: PKPaymentRequest
+    
+    /// The last status of the authorization received by passing the token data
+    /// to the payment server
+    var lastAuthStatus: PKPaymentAuthorizationStatus!
+    
+    var summaryItems: [PKPaymentSummaryItem]
+    var shippingMethods: [PKShippingMethod]
+    weak var delegate: PaymentSessionDelegate?
+    
+    // ----------------------------------
+    //  MARK: - Init -
+    //
+    init(withMerchantIdentifier merchantId: String, countryCode: String, currencyCode: String, requiringShippingAddressFields: Bool)
+    {
+        summaryItems    = [PKPaymentSummaryItem]()
+        shippingMethods = [PKShippingMethod]()
+        
+        request = PKPaymentRequest.init()
+        request.countryCode                   = countryCode
+        request.currencyCode                  = currencyCode
+        request.merchantIdentifier            = merchantId
+        request.requiredBillingAddressFields  = .all
+        request.merchantCapabilities          = .capability3DS
+        request.supportedNetworks             = PaymentSession.supportedNetworks
+        request.requiredShippingAddressFields = requiringShippingAddressFields ?
+            .all : PKAddressField.init(rawValue: PKAddressField.email.rawValue | PKAddressField.phone.rawValue)
+        
+        super.init()
+    }
+    
+    func addSummaryItem(withLabel label : String, amount: NSDecimalNumber) {
+        let newItem = PKPaymentSummaryItem.init(label: label, amount: amount)
+        summaryItems.append(newItem)
+    }
+    
+    func removeAllSummaryItems() {
+        summaryItems.removeAll()
+    }
+    
+    func addShippingMethod(withLabel label : String, amount: NSDecimalNumber, identifier: String, detail: String) {
+        let newMethod        = PKShippingMethod.init(label: label, amount: amount)
+        newMethod.identifier = identifier
+        newMethod.detail     = detail
+        
+        shippingMethods.append(newMethod)
+    }
+    
+    func removeAllShippingMethods() {
+        shippingMethods.removeAll()
+    }
+    
+    /// Assigns the shipping methods and summary items in this session to the request
+    func populateRequest() {
+        request.shippingMethods     = shippingMethods;
+        request.paymentSummaryItems = summaryItems;
+    }
+    
+    /// Presents the PKAuthorizationController with the current shipping methods
+    /// and summary items in this session.
+    /// UnityAppController is set to remain active until the authorization controller
+    /// is dismissed, to enable communication between managed and unmanaged functions
+    func presentAuthorizationController()  {
+        populateRequest();
+        let authViewController = PKPaymentAuthorizationViewController.init(paymentRequest: request)
+        let unityController    = UIApplication.shared.delegate as! UnityBuyAppController
+        let topController      = unityController.topMostController()!
```

Comment:
Alternatively, you can present this on the window's root view controller:
```swift
unityController.window.rootViewController
```
The decision is up to you.